### PR TITLE
OSPK8-364 use j2 template for rendering osnet CRs

### PR DESCRIPTION
**Pull request by stuggi**
Right now we have multiple temple files for the OSP networks.
This PR move to have a single j2 template and now it is possible
to specify networks using vars.

Also to not have to copy the full osp dict from var/defaults.yaml to
the version specific var file, this switches to use osp_defaults
dict in var/defaults.yaml and allows to overwrite specific parts
from the dict using osp_release_defaults dict in the version
specific, but also osp_local in local-defaults.yaml vars file.
Where required the combined osp dict gets created in the
playbook where it is currently used.

### CODE REVIEW

File: ansible/vars/16.2.yaml [hunk 1]
```diff
@@ -1,34 +1 @@
...
-  interface:
-    external: enp6s0
-    ospnetwork: enp7s0
-  network_type: vlan
-  container_tag: 16.2_20210901.2
-  ceph_image: rhceph
-  ceph_tag: 5-12
-  # OSP deployment timeout
-  deploy_timeout: 60m
+osp: {}
```
Comment (olliewalsh) [inline ansible/vars/16.2.yaml:1]: Could this be renamed to `osp_release_defaults` and use `osp` for the combined fact?

File: ansible/datavolume.yaml [hunk 2]
```diff
@@ -6,21 +6,24 @@
   - oc_local
 
   tasks:
+  - name: Set combined osp dict
+    set_fact:
+      osp_combined: "{{ osp_defaults | combine((osp | default({})), recursive=True) }}"
```
Comment (olliewalsh) [inline ansible/datavolume.yaml:11]: Could we combine with an `osp_local` fact too that could be set in local-defaults?

File: ansible/templates/osp/ctlplane/osp-director_openstackcontrolplane.yaml.j2 [hunk 3]
```diff
@@ -10,21 +10,21 @@ spec:
...
         - internalapi
-  openStackClientStorageClass: {{ osp.controller.storage_class }}
+  openStackClientStorageClass: {{ osp_combined.controller.storage_class }}
   passwordSecret: userpassword
   virtualMachineRoles:
     controller:
       roleName: Controller
-      roleCount: {{ osp.controller.count }}
+      roleCount: {{ osp_combined.controller.count }}
       networks:
```
Comment (olliewalsh) [inline ansible/templates/osp/ctlplane/osp-director_openstackcontrolplane.yaml.j2:19]: Should this be templated too?

File: ansible/vars/16.2.yaml (hunk 1 above)
Comment (stuggi) [reply ansible/vars/16.2.yaml:1]: you mean osp_releae_defaults in defaults.yaml, osp_release in the version specific overwrites and then osp as the combined to be used by the templates. That makes sense and I'd not have to change all the templates :)

File: ansible/datavolume.yaml (hunk 2 above)
Comment (stuggi) [reply ansible/datavolume.yaml:11]: ack good idea!

File: ansible/templates/osp/ctlplane/osp-director_openstackcontrolplane.yaml.j2 (hunk 3 above)
Comment (stuggi) [reply ansible/templates/osp/ctlplane/osp-director_openstackcontrolplane.yaml.j2:19]: yes for flexibility that makes sense. I'll do this in a follow up when working on subnet support

File: ansible/vars/16.2.yaml (hunk 1 above)
Comment (stuggi) [reply ansible/vars/16.2.yaml]: done

File: ansible/datavolume.yaml (hunk 2 above)
Comment (stuggi) [reply ansible/datavolume.yaml]: done

**Review by olliewalsh — COMMENTED** [review]
/lgtm

**Review by abays — APPROVED** [review]
/lgtm

**Pull request merged**